packages/payment-detection/test/any/any-to-erc20-proxy-contract.test.ts

```diff
@@ -0,0 +1,303 @@
+      paymentAddress: 'ethereum address',
+    });
+
+    expect(spy).toHaveBeenCalledWith({
+      paymentAddress: 'ethereum address',
+    });
+  });
+
+  it('can createExtensionsDataForAddRefundInformation', async () => {
+    const spy = jest.spyOn(
+      mockAdvancedLogic.extensions.anyToErc20Proxy,
+      'createAddRefundAddressAction',
+    );
+
+    anyToErc20Proxy.createExtensionsDataForAddRefundInformation({
+      refundAddress: 'ethereum address',
+    });
+
+    expect(spy).toHaveBeenCalledWith({
+      refundAddress: 'ethereum address',
+    });
+  });
+
+  it('can createExtensionsDataForAddFeeInformation', async () => {
+    const spy = jest.spyOn(
+      mockAdvancedLogic.extensions.anyToErc20Proxy,
+      'createAddFeeAction',
+    );
+
+    anyToErc20Proxy.createExtensionsDataForAddFeeInformation({
+      feeAddress: 'ethereum address',
+      feeAmount: '2000',
+    });
+
+    expect(spy).toHaveBeenCalledWith({
+      feeAddress: 'ethereum address',
+      feeAmount: '2000',
+    });
+  });
+
+  it('should not throw when getBalance fail', async () => {
+    expect(
+      await anyToErc20Proxy.getBalance({ extensions: {} } as RequestLogicTypes.IRequest),
+    ).toEqual({
+      balance: null,
+      error: {
+        code: PaymentTypes.BALANCE_ERROR_CODE.WRONG_EXTENSION,
+        message: 'The request does not have the extension : pn-any-to-erc20-proxy',
+      },
+      events: [],
+    });
+  });
+
+  it('can get the fees out of payment events', async () => {
+    const mockRequest: RequestLogicTypes.IRequest = {
+      creator: { type: IdentityTypes.TYPE.ETHEREUM_ADDRESS, value: '0x2' },
+      currency: {
+        type: RequestLogicTypes.CURRENCY.ISO4217,
+        value: 'EUR',
+      },
+      events: [],
+      expectedAmount: '1000',
+      extensions: {
+        [ExtensionTypes.ID.PAYMENT_NETWORK_ANY_TO_ERC20_PROXY]: {
+          events: [],
+          id: ExtensionTypes.ID.PAYMENT_NETWORK_ANY_TO_ERC20_PROXY,
+          type: ExtensionTypes.TYPE.PAYMENT_NETWORK,
+          values: {
+            feeAddress: '0xC5fdf4076b8F3A5357c5E395ab970B5B54098Fef',
+            feeAmount: '5',
+            paymentAddress: '0xf17f52151EbEF6C7334FAD080c5704D77216b732',
+            refundAddress: '0x666666151EbEF6C7334FAD080c5704D77216b732',
+            acceptedTokens: ['0x9FBDa871d559710256a2502A2517b794B482Db40'],
+          },
+          version: '0',
+        },
+      },
+      extensionsData: [],
+      requestId: '0x1',
+      state: RequestLogicTypes.STATE.CREATED,
+      timestamp: 0,
+      version: '0.2',
+    };
+
+    const mockExtractBalanceAndEvents: any = (
+      _request: RequestLogicTypes.IRequest,
+      _salt: string,
+      _toAddress: string,
+      eventName: PaymentTypes.EVENTS_NAMES) => {
+      if(eventName === PaymentTypes.EVENTS_NAMES.PAYMENT) {
+        return Promise.resolve({
+          balance: '1000',
+          events: [
+            // Wrong fee address
```

Comment:
Isn't it more relevant with this payment network to test the payment in a token that is not accepted? It's the tricky part we are adding.

Reply:
This part is made in the info-retriever. 
and it's tested by `skips the payment with a payment token that is not accepted` in `any-to-erc20-proxy-info-retriever.test.ts`.